### PR TITLE
[AMQ-8345] Update JMS2 progress page

### PR DESCRIPTION
**Pull request by mattrpav**
- Remove RC notation, as there is no planned "RC"-style release
 - Update tracking based on planned version alignment to JIRA tasks

### CODE REVIEW

**Comment by gemmellr** [comment]
I'd like to suggest this again:
> JB previously suggested he wanted to move the component specific docs etc into the component area, as part of a general tidyup and simplification of the docs. While the overall effort doesnt seem to have progressed as yet, as a rather component-specific page and probably the first distinct new page I can think of being created in a while, I think it would probably make sense to start the trend with this?

https://github.com/apache/activemq-website/pull/54#issuecomment-911633870

Personally I think retaining the RC notion would also make sense given the suggested state of the implementation.

File: src/jms2.md [hunk 1]
```diff
@@ -25,11 +25,13 @@ User feedback is welcome! Please comment on the JIRAs with questions and comment
...
-[AMQ-8324](https://issues.apache.org/jira/browse/AMQ-8324) | 5.17.0.RC1 | | MessageProducer send methods | Completion Listener and Delivery Delay support
+[AMQ-7309](https://issues.apache.org/jira/browse/AMQ-7309) | 5.17.0 | | JMS v2.0 API dependency | ActiveMQ will ship with a JMS v2.0 dependency jar
+[AMQ-8322](https://issues.apache.org/jira/browse/AMQ-8322) | 5.17.0 | | Connection.createContext | Simplified JMS Context object support
+[AMQ-8322](https://issues.apache.org/jira/browse/AMQ-8322) | 5.17.0 | | Session Topic Consumer | Consume messages using a MessageConsumer object (same as Queue)
+[AMQ-8322](https://issues.apache.org/jira/browse/AMQ-8322) | 5.17.0 | | MessageProducer send methods | 
+[AMQ-8320](https://issues.apache.org/jira/browse/AMQ-8320) | 5.17.1 | | Delivery Delay | Support for Message DeliveryDelay feature
+[AMQ-8321](https://issues.apache.org/jira/browse/AMQ-8321) | 5.17.1 | | GetBody/isBodyAssignable | Support for checking body type using a Class<?>
+[AMQ-8324](https://issues.apache.org/jira/browse/AMQ-8324) | TBD | | MessageProducer send methods | Completion Listener and Delivery Delay support
```
Comment (gemmellr) [inline src/jms2.md:34]: First "MessageProducer send methods" line has a terminator when no other does.

Seems odd to have 2 lines and JIRAs referenced for "MessageProducer send methods" as a feature, when the first is unqualified with notes. If its incomplete it should probably be noted in the first instance. Especially as the second has no timeline given and its notes mention the main send feature additions in JMS 2.

Comment (mattrpav) [reply src/jms2.md]: I re-worked the table to clarify the intent

**Comment by mattrpav** [comment]
image

**Pull request merged**